tools/test-proxy/Azure.Sdk.Tools.TestProxy/RecordingHandler.cs

```diff
@@ -464,8 +467,67 @@ public async Task HandlePlaybackRequest(string recordingId, HttpRequest incoming
             {
                 var bodyData = CompressionUtilities.CompressBody(match.Response.Body, match.Response.Headers);
 
-                outgoingResponse.ContentLength = bodyData.Length;
+                if (match.Response.Headers.ContainsKey("Content-Length"))
+                {
+                    outgoingResponse.ContentLength = bodyData.Length;
+                }
+
+                await WriteBodyBytes(bodyData, session.PlaybackResponseTime, outgoingResponse);
+            }
+        }
+
+        public byte[][] GetBatches(byte[] bodyData, int batchCount)
```

Comment:
Perf could be improved to avoid copying bytes by returning something like `IEnumerable<ReadOnlyMemory<byte>>` instead of `byte[][]`, but I don't think many requests will use the `playbackResponseTime` feature, and certainly not perf-sensitive users, so the current implementation is fine.